tests/test_books.py

```diff
@@ -57,3 +57,29 @@ def test_user_profile_page(client, user, user_books, snippet):
     response = client.get(f'/users/{user.username}')
     html = response.content.decode()
     assert snippet in html
+
+
+def test_user_profile_page_no_favorites(client, user):
+    response = client.get(f'/users/{user.username}')
+    html = response.content.decode()
+    assert 'Favorite Books' not in html
+
+
+def test_user_profile_page_favorites(client, user, user_fav_books):
+    response = client.get(f'/users/{user.username}')
+    html = response.content.decode()
+    assert 'Favorite Books' in html
+
+
+@pytest.mark.parametrize("snippet, is_in", [
+    ("bookid=nneBa6-mWfgC  checked", False),
```

Comment:
Interesting, in Chrome dev tools > inspect I see one space, you can match the whole input html snippet, then you don't need the 2nd boolean argument, this is also more specific:

<img width="1179" alt="image" src="https://user-images.githubusercontent.com/387927/97105727-73a8bb80-16bd-11eb-8b7e-91bf0547bf14.png">

Reply:
Yes this is a bit odd  which is why I mentioned it. 
I see two spaces if I view the page source:

![image](https://user-images.githubusercontent.com/45608909/97112356-b8dbe600-16db-11eb-9c52-754248bb7b2a.png)

But it looks like a single space in developer tools (This from Chrome on Win10):

![image](https://user-images.githubusercontent.com/45608909/97112390-ed4fa200-16db-11eb-9775-e1bfacf170a1.png)

I've used two spaces in the tests and they run OK.

I've matched the whole input snippet as you suggest, but in the interests of DRY I've used an F String with the common content and just passed in the varying checked content. 
If you prefer to see the whole input in the snippet for clarity let me know and I'll change it